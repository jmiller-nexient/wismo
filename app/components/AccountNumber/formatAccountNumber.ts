export default function formatAccountNumber(value: string) {

  const formattedAccountNumber = `${value.substr(0,4)} ${value.substr(4,3)} ${value.substr(7,4)} ${value.substr(11,1)}`;

  return formattedAccountNumber;
}
