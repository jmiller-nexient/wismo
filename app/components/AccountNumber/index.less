@import '../../styles/themes/index.less';

.dte-wismo-account-number {
  color: @dte-gray-500;
  font-family: @dte-wismo-font-family;
  font-size: @dte-wismo-font-size-1;
}
