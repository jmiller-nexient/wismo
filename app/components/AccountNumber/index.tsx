import * as React from 'react';

import formatAccountNumber from './formatAccountNumber';

interface IAccountNumberProps {
  value: string;
}

const AccountNumber: React.FC<IAccountNumberProps> = (props: IAccountNumberProps): React.ReactElement => {
  const { value } = props;

  if (value.length !== 12) {
    throw new Error('account numbers must be 12 numeric digits.');
  }

  const formattedAccountNumber = formatAccountNumber(value);

  return <span>{formattedAccountNumber}</span>;
};

export default AccountNumber;
