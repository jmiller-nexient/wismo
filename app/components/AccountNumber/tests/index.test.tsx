import React, { ReactElement } from 'react';

import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

import AccountNumber from '..';

describe('<AccountNumber />', () => {
  it('should render correctly', () => {
    const tree: ReactElement = renderer.create(
      <AccountNumber value={'112233445566'} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should throw an error when account number is less than 12 digits', () => {
    expect(() => shallow(<AccountNumber value={'123456'}/>)).toThrowError();
  });

  it('should throw an error when account number is more than 12 digits', () => {
    expect(() => shallow(<AccountNumber value={'1234561234567'}/>)).toThrowError();
  });
});
