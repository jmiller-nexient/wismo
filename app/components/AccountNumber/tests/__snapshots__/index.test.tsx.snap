// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountNumber /> should render correctly 1`] = `
<span>
  1122 334 4556 6
</span>
`;
